### PR TITLE
Plugin version v7

### PR DESCRIPTION
**Pull request by bureddy**
removed old v4 version

### CODE REVIEW

File: include/net.h [hunk 1]
```diff
@@ -22,10 +22,8 @@ typedef enum {NCCL_INIT=1, NCCL_COLL=2, NCCL_P2P=4, NCCL_SHM=8, NCCL_NET=16, NCC
 
 typedef void (*ncclDebugLogger_t)(ncclDebugLogLevel level, unsigned long flags, const char *file, int line, const char *fmt, ...);
 
+#include "net_v7.h"
 #include "net_v6.h"
 #include "net_v5.h"
-#include "net_v4.h"
-
-#define NCCL_PLUGIN_SYMBOL ncclNetPlugin_v6
```
Comment (wenbilliams) [inline include/net.h:29]: Should this be redefined to v7?

File: src/p2p_plugin.c [hunk 2]
```diff
@@ -40,30 +40,31 @@ extern int ncclIbRelaxedOrderingEnabled;
 NCCL_PARAM(SharpMaxComms, "SHARP_MAX_COMMS", 1);
 NCCL_PARAM(IbAdaptiveRouting, "IB_ADAPTIVE_ROUTING", -2);
 
-ncclResult_t pluginInit(ncclDebugLogger_t logFunction);
+ncclResult_t pluginInit_v7(ncclDebugLogger_t logFunction);
+ncclResult_t pluginInit_v6(ncclDebugLogger_t logFunction);
+ncclResult_t pluginInit_v5(ncclDebugLogger_t logFunction);
+
+ncclNet_v7_t ncclNetPlugin_v7 = {
+  "NCCL RDMA Plugin v7",
```
Comment (wenbilliams) [inline src/p2p_plugin.c:48]: Do you think this name should be more descriptive? I see this is following the old pattern

File: src/sharp_plugin.c [hunk 3]
```diff
@@ -123,8 +123,8 @@ int ncclSharpAllGather(void *context, void *buf, int len) {
 
   p2p_plugin = nccl_p2p_get_plugin_type();
   if (p2p_plugin != NCCL_P2P_UCX) {
-    NCCLCHECK(NCCL_PLUGIN_SYMBOL.regMr(cComm->recvComm, buf, cComm->nranks*len, NCCL_PTR_HOST, &rMhandle));
```
Comment (wenbilliams) [inline src/sharp_plugin.c:126]: Ah I see, you just want to be explicit. Makes sense.

File: src/ucx_plugin.c [hunk 4]
```diff
@@ -888,58 +920,19 @@ ncclNet_v5_t ucxPlugin_v5 = {
...
-  return ret;
-}
-
-ncclNet_v4_t ucxPlugin_v4 = {
+ncclNet_v5_t ucxPlugin_v5 = {
   .name = "UCX",
   .init = nccl_ucx_init,
   .devices = nccl_ucx_devices,
-  .getProperties = nccl_ucx_get_properties_v4,
+  .getProperties = nccl_ucx_get_properties_v6,
```
Comment (wenbilliams) [inline src/ucx_plugin.c:927]: Should this be v5 or are you reusing v6 for v5 when it hasn't changed?

File: src/p2p_plugin.c (hunk 2 above)
Comment (bureddy) [reply src/p2p_plugin.c:48]: this is a temporary name. It is overwritten when the actual plugin (ib/ucx) is assigned in line no 88.

File: src/ucx_plugin.c (hunk 4 above)
Comment (bureddy) [reply src/ucx_plugin.c:927]: reusing

**Pull request merged**